src/main/cc/proxy_client/javac_remote_actions.cc

```diff
@@ -11,15 +15,74 @@ const int kMaxArgsToCheckForJavac = 8;
 const char* kJavacWrapperCommand = "javac_wrapper ";
 const char* kJavacCommand = "javac ";
 
+const char* kJavacWithOptionsCommand = "javac -";
+
 using std::min;
+using std::string;
+using std::set;
+using std::cout;
+
+bool IsAlphabet(const char c) {
+  return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z');
+}
+
+string Trim(const string& cmd) {
+  if (cmd.length() == 0) {
+    return "";
+  }
+
+  size_t st_idx = 0;
+  size_t ed_idx = cmd.length() - 1;
+
+  while (st_idx <= ed_idx && !IsAlphabet(cmd[st_idx])) ++st_idx;
+  while (ed_idx > st_idx && !IsAlphabet(cmd[ed_idx])) --ed_idx;
+
+  return cmd.substr(st_idx, ed_idx - st_idx + 1);
+}
+
+bool IsFile(const string& name) {
+  struct stat buffer;   
+  return (stat (name.c_str(), &buffer) == 0);
```

Comment:
Style nitpick remove space between stat and "("

Reply:
Done.